libwacom/libwacom.c

```diff
@@ -1362,7 +1362,17 @@ libwacom_get_height(const WacomDevice *device)
 LIBWACOM_EXPORT WacomClass
 libwacom_get_class(const WacomDevice *device)
 {
-	return device->cls;
+	if (device->cls != WCLASS_UNKNOWN) {
+		return device->cls;
+	}
+	if (device->integration_flags & WACOM_DEVICE_INTEGRATED_DISPLAY &&
+		device->integration_flags & WACOM_DEVICE_INTEGRATED_SYSTEM) {
+		return WCLASS_ISDV4;
+	}
+	if (device->integration_flags & WACOM_DEVICE_INTEGRATED_DISPLAY) {
+		return WCLASS_CINTIQ;
+	}
+	return WCLASS_BAMBOO;
```

Comment:
I think the `Remote` class is missing here, that will need handling, could go together with #779.

Then again there's so few of them that we can mark those in the `.tablet` file...

Reply:
Oh, come to think of it: this is better as `WCLASS_INTUOS5` - that's the most recent class and most devices are more like that than whatever the latest bamboo was like.

Reply:
For the archives: `WCLASS_BAMBOO` is better here because bamboos arent' expected to have stylus ids/serials.